src/addrdb.cpp

```diff
@@ -119,18 +176,49 @@ bool DeserializeFileDB(const fs::path& path, Data& data)
 }
 } // namespace
 
-CBanDB::CBanDB(fs::path ban_list_path) : m_ban_list_path(std::move(ban_list_path))
+CBanDB::CBanDB(fs::path ban_list_path)
+    : m_banlist_dat(ban_list_path.string() + ".dat"),
+      m_banlist_json(ban_list_path.string() + ".json")
 {
 }
 
 bool CBanDB::Write(const banmap_t& banSet)
 {
-    return SerializeFileDB("banlist", m_ban_list_path, banSet);
+    std::vector<std::string> errors;
+    if (util::WriteSettings(m_banlist_json, {{JSON_KEY, BanMapToJson(banSet)}}, errors)) {
+        return true;
+    }
+
+    for (const auto& err : errors) {
+        error("%s", err);
+    }
+    return false;
 }
 
 bool CBanDB::Read(banmap_t& banSet)
 {
-    return DeserializeFileDB(m_ban_list_path, banSet);
+    if (!fs::exists(m_banlist_json)) {
+        return DeserializeFileDB(m_banlist_dat, banSet);
```

Comment:
51909ded1e8b26158f8d36cfe323ac72feac1d78

Maybe migrate on 1st run?
```diff
 bool CBanDB::Read(banmap_t& banSet)
 {
     if (!fs::exists(m_banlist_json)) {
-        return DeserializeFileDB(m_banlist_dat, banSet);
+        if (!DeserializeFileDB(m_banlist_dat, banSet)) {
+            return false;
+        }
+        // log something
+        Write(banSet);
+        return true;
     }
```

Reply:
Migrate on first run - yes. But `Read()` calling `Write()` seems like some anti-pattern to me. Instead I signal from `Read()` that the list needs re-flushing, so that the caller takes care of it.

Reply:
Good point. Maybe add a `CBanDB::MigrateAndRead` where calling `Write()` isn't so bad.

Reply:
Not necessary with the latest changes.